### ISSUE TITLE
Do not match constants or literals

### ISSUE THREAD

**Issue by vanillajonathan** [rust-lang/rust-clippy#11224]
### What it does

Checks if you're using `match` against a constant or a literal.

### Advantage

There is no need to perform a match against a constant.

### Drawbacks

Maybe it would raise false positives for compile-time constants.

### Example

```rust
const FOO: i32 = 3;
let bar = match FOO {
    1 => "true",
    _ => "false",
};
```
or:
```rust
let bar = match 3 {
    1 => "true",
    _ => "false",
};
```

Could be written as:

```rust
let bar = "true"
```

**Comment by Centri3** [comment]
This should ignore `cfg`-dependent constants. Great idea, though!
